### PR TITLE
test(sdk/python): Add comprehensive pytest-based tests for Python SDK

### LINKED ISSUE #44: Add pytest-based tests for Python SDK
## Request
Create pytest-based tests for the Python SDK to ensure functionality and improve code reliability.

### Details
- Develop tests for core modules and functionality in the Python SDK.
- Use pytest as the testing framework for consistency.
- Ensure coverage for existing and major new features.
- Integrate tests to run in the CI pipeline if not already applied.

### Acceptance Criteria
- All critical components and features of the Python SDK are covered by pytest-based tests.
- Tests pass successfully in the CI pipeline.

---
Feel free to discuss or break this into sub-issues as needed.

### PR DESCRIPTION
**Pull request by PredictiveManish**
## Summary

Add comprehensive pytest-based tests for the Python SDK to ensure functionality and improve code reliability. This PR adds 39 new unit tests covering previously untested functionality including `unload_index`, cloud fallback error handling, URL helpers, data type usage, and more.

Additionally, fixes a bug where `alpha=0` was incorrectly treated as falsy in query options, preventing pure keyword search from working correctly.

Checklist:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) guide.
- [ ] I have updated the documentation (if applicable).
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

### New Test Files

**`tests/test_client_extended.py`** (38 new tests)
- `TestUnloadIndex` - Tests for the `unload_index` method
- `TestCloudFallbackErrors` - HTTP errors, network errors, custom embeddings in cloud fallback
- `TestDictToSearchResult` - Edge cases for search result dictionary conversion
- `TestURLHelpers` - Environment variable URL configuration testing
- `TestVersionExport` - Version string validation
- `TestQueryOptionsBehavior` - Alpha=0 keyword-only search behavior
- `TestMetadataFilterWarning` - Filter warning when index is not loaded
- `TestClientIdGeneration` - UUID generation for client instances
- `TestProjectCredentials` - Credential handling

**`tests/test_types.py`** (11 new tests)
- `TestQueryOptionsUsage` - Query options in client operations
- `TestGetDocumentsOptionsUsage` - Document retrieval options
- `TestMutationOptionsUsage` - Mutation options (upsert mode)
- `TestDocumentInfoUsage` - Document creation with metadata/embeddings
- `TestSearchResultUsage` - Search result structure validation
- `TestIndexInfoUsage` - Index info handling
- `TestJobStatusResponseUsage` - Job status response handling

### Bug Fix

**`src/moss/client/moss_client.py:204-209`**
- Changed `top_k = getattr(options, "top_k", None) or 5` to explicit None check
- Changed `alpha = getattr(options, "alpha", None) or 0.8` to explicit None check
- **Issue**: Using `or` operator treated `0` as falsy, preventing pure keyword search (`alpha=0`) from working

## Testing

```bash
# Run unit tests
pytest tests/test_client.py tests/test_client_extended.py tests/test_types.py -v

# Run linting
ruff check src/ tests/

# Run type checking
mypy src/ --ignore-missing-imports
```

## Test Results
- 86 tests pass (47 existing + 39 new)
- Ruff linting: All checks passed
- Mypy type checking: Success

Fixes #44 
## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: sdks/python/sdk/tests/test_client_extended.py [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+    """Tests for version export."""
+
+    def test_version_is_defined(self):
+        assert __version__ is not None
+        assert isinstance(__version__, str)
+        assert len(__version__) > 0
+
+    def test_version_format(self):
+        assert __version__ == "1.0.0b19"
+
```
[the remark this answers is not in the document]
Comment (PredictiveManish) [reply sdks/python/sdk/tests/test_client_extended.py:251]: Check again I have added 
```
from importlib.metadata import version

def test_version_format(self):
    assert __version__ == version("moss")
```

File: sdks/python/sdk/tests/test_types.py [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+            ]
+            result = await client.create_index("idx", docs)
+
+            assert result.job_id == "j1"
+            mock_manage_instance.create_index.assert_called_once()
+
+    @pytest.mark.asyncio
+    async def test_create_index_with_metadata(self):
+        with (
+            patch("moss.client.moss_client.ManageClient") as mock_manage,
```
Comment (HarshaNalluru) [inline sdks/python/sdk/tests/test_types.py:154]: re-creates the client with inline patch() context managers in nearly every test class. Consider extracting a shared fixture in conftest.py to cut the boilerplate.

Comment (PredictiveManish) [reply sdks/python/sdk/tests/test_types.py:154]: updated!

**Review by HarshaNalluru — APPROVED** [review]
